bin/add_annotsv.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+from pysam import VariantFile
+from pprint import pprint
+import cmdvcf
+import argparse
+
+def main(args):
+    """
+    main function, reads VCF line by line and adds all AnnotSV annotations
+    Updates vcf-header with the new annotations being added
+    """
+    annotsv_variants, annotsv_headers = read_annotsv_vcf(args.annotsv)
+    vcf_object = VariantFile(args.input_vcf)
+    
+    vcf_header = join_headers(vcf_object.header,annotsv_headers)
+    vcf_out = VariantFile(args.out_vcf, "w", header=vcf_header)
+    vcf_out.close()
+    for var in vcf_object.fetch():
+        # to fix END flag disapearing from pysam object
+        # pysam eats up END from original, don't know why. (PERL vfc2.pm <3)
+        var_dict = cmdvcf.parse_variant(var,vcf_object.header)
+        var_dict['INFO']['END'] = var.stop
+        svtype = var.info['SVTYPE']
+        # AnnotSV treats TDUP as DUP
+        if svtype == "TDUP":
+            svtype = "DUP"
+        simple_id = f"{var.chrom}_{var.pos}_{svtype}"
+        if simple_id in annotsv_variants:
+            var_dict['INFO'] = add_annotsv_annotations(var_dict['INFO'],annotsv_variants[simple_id]['INFO'])
+        vcf_str = cmdvcf.print_vcf_record(var_dict,vcf_object.header)
+        with open(args.out_vcf, "a") as vcf_out:
+            vcf_out.write(vcf_str)
+        
+def add_annotsv_annotations(original_info,annotation):
+    """
+    Add the unique(new) keys from annotsv to original VCF INFO
+    """
+    for key in annotation:
+        if key not in original_info:
+            original_info[key] = annotation[key]
+    return original_info
+
+def join_headers(original_header, new_annotations):
+    """
+    add new annotation headers to INFO
+    """
+    for annotation in new_annotations:
+        if annotation['key'] not in original_header.info:
+            original_header.info.add(
+                annotation['key'],
+                annotation['number'],
+                annotation['type'],
+                annotation['description']
+            )
+    return original_header
+
+def read_annotsv_vcf(file_path):
+    """
+    read VCF from annotsv
+    """
+    vcf_in = VariantFile(file_path, "r")
```

Comment:
Do you need to close it again?

I think you can do a scoped opening here as well. Then you don't need to worry about remembering closing it.

```
with VariantFile(file_path, "r") as vcf_in:
  ...
```